networkx/generators/classic.py

```diff
@@ -706,6 +707,56 @@ def star_graph(n, create_using=None):
     return G
 
 
+@nodes_or_number([0, 1])
+@nx._dispatch(graphs=None)
+def tadpole_graph(m, n, create_using=None):
+    """Returns the (m,n)-tadpole graph.
+
+    This graph on m+n nodes connects a cycle of size m to a path of length n.
+    It looks like a tadpole.
+    It is also called a kite graph or a dragon graph.
+
+    Parameters
+    ----------
+    m, n : int or iterable container of nodes (default = 0)
+        If an integer, nodes are from `range(m)` and `range(m,m+n)`.
+        If a container of nodes, those nodes appear in the graph.
+        Warning: m and n are not checked for duplicates and if present the
+        resulting graph may not be as desired.
+
+        The nodes for m appear in the cycle graph $C_m$ and the nodes
+        for n appear in the path $P_n$
+    create_using : NetworkX graph constructor, optional (default=nx.Graph)
+       Graph type to create. If graph instance, then cleared before populated.
+
+    Notes
+    -----
+    The 2 subgraphs are joined via an edge (m-1, m).
+    If n=0, this is a cycle graph.
+    If m=0, this is a path graph.
```

Comment:
```suggestion
    If m=0 or 2, this is a path graph. (m=1 gives a path with a self-loop at one end)
    m and/or n can be a container of nodes instead of an integer.
```